Crucible/SchemaCore/SchemaTranslators/ClassTranslator.cs

```diff
@@ -0,0 +1,227 @@
+  public class ClassTranslator : ISchemaTranslator<object>
+  {
+    /// <inheritdoc/>
+    public bool CollectionContains(object collection, string valueName) => collection.GetType().GetProperties().Select(x => x.Name).Contains(valueName);
+    /// <inheritdoc/>
+    public string CollectionValueToString(object collection, string valueName)
+    {
+      PropertyInfo valueProperty = collection.GetType().GetProperty(valueName);
+      return valueProperty.GetValue(collection).ToString();
+    }
+    /// <inheritdoc/>
+    public List<string> GetCollectionKeys(object collection) => collection.GetType().GetProperties().Select(x => x.Name).ToList();
+    /// <inheritdoc/>
+    public string GetEquivalentType(string cSharpType) => Conversions.GetEquivalentJsonType(cSharpType);
+
+    /// <summary>
+    /// Inserts <paramref name="newValue"/> into <paramref name="collection"/>
+    /// with string designation <paramref name="valueName"/> and returns a
+    /// new instance of the class object with the value inserted.
+    /// </summary>
+    /// <exception cref="ArgumentException">If <paramref name="collection"/>'s
+    /// base type does not contain a property named <paramref name="valueName"/>
+    /// or if it does contain a property of that name, but the type is not
+    /// compatible with <typeparamref name="TDefaultValueType"/></exception>
+    /// <typeparam name="TDefaultValueType">Type of the value to be inserted.</typeparam>
+    /// <param name="collection">The collection from which to extract the value.</param>
+    /// <param name="valueName">The string designator for the value to be extracted.</param>
+    /// <param name="newValue">The new value to be inserted.</param>
+    /// <returns>New object with value inserted.</returns>
+    public object InsertFieldValue<TDefaultValueType>(object collection, string valueName, TDefaultValueType newValue)
+    {
+      try
+      {
+        PropertyInfo valueProperty = collection.GetType().GetProperty(valueName);
+        if(valueProperty.PropertyType == typeof(TDefaultValueType))
+        {
+          valueProperty.SetValue(collection, newValue);
+        }
+        else
+        {
+          valueProperty.SetValue(collection, Convert.ChangeType(newValue, valueProperty.PropertyType));
+        }
+        return collection;
+      }
+      catch
+      {
+        throw new ArgumentException($"Attempted to set property {valueName} on class type {collection.GetType().FullName} to value with type {typeof(TDefaultValueType).FullName}");
+      }
+    }
+    /// <inheritdoc/>
+    public bool FieldValueIsNullOrEmpty(object collection, string valueName)
+    {
+      PropertyInfo valueProperty = collection.GetType().GetProperty(valueName);
+      Type valueType = valueProperty.PropertyType;
+      object value = valueProperty.GetValue(collection);
+      return value != null
+        && (valueType.IsAssignableFrom(typeof(JToken)) ? ((JToken)value).IsNullOrEmpty() : value.ToString().IsNullOrEmpty());
```

Comment:
Don't check against `ToString()` unless you can guarantee that it's implemented in way you expect in all cases.

Reply:
If `value` is a string, is whitespace considered empty in the way you expect?

Reply:
Ignore previous comment, lol. I'm thinking of a different extension (`IsNullOrWhiteSpace`). You're good.